### PR TITLE
move packages to cmd, update instructions

### LINKED ISSUE #42: simplify install
`go` supports cloning via git+ssh, so you can use the toolchain to do the install ie. `go install github.com/francoismichel/ssh3/cli/server@latest`. You need better folder names though, b/c it's going to install the tool as `$GOBIN/server`

In go, it's more typical to use a `cmd` sub folder, maybe checkout go tools for ideas on how you want to support it: https://pkg.go.dev/golang.org/x/tools

My suggestion would be something like:

Server install
```
go install github.com/francoismichel/ssh3/cmd/sshd@latest
```

Client install
```
go install github.com/francoismichel/ssh3/cmd/ssh@latest
```

I'm happy to reoganize the repo to pretty paths.

### PR DESCRIPTION
**Pull request by drewwells**
fixes #42

### CODE REVIEW

**Comment by francoismichel** [comment]
You may want to know that the server does not compile on MacOS, it's being worked out in #56 but it is not part of main yet.

**Comment by drewwells** [comment]
> You may want to know that the server does not compile on MacOS, it's being worked out in #56 but it is not part of main yet.
> 
> So right now let's only build the client on these platforms and we'll update the install once #56 gets merged (very soon I think)

Good to know, updated

File: README.md [hunk 1]
```diff
@@ -64,12 +64,18 @@ This SSH3 implementation already provides many of the popular features of OpenSS
...
 or generate these binaries yourself by compiling the code from source.
 
 > [!TIP]
 > SSH3 is still experimental and is the fruit of a research work. If you are afraid of deploying publicly a new SSH3 server, you can use the
-> [secret path](#-your-ssh3-public-server-can-be-hidden) feature of SSH3 to hide it behing a secret URL. 
+> [secret path](#-your-ssh3-public-server-can-be-hidden) feature of SSH3 to hide it behing a secret URL.
+
+### Installing ssh3 and ssh3-server
+```bash
+git install https://github.com/francoismichel/ssh3/cmd/...@latest
```
Comment (francoismichel) [inline README.md:76]:
```suggestion
go install https://github.com/francoismichel/ssh3/cmd/...@latest
```

**Comment by francoismichel** [comment]
Merged in [5cb4b19](https://github.com/francoismichel/ssh3/commit/5cb4b197e5c06c1e87124b88cba8005097ed2858), thanks a lot for the help !

**Pull request merged**